management_api_app/db/repositories/workspaces.py

```diff
@@ -17,27 +17,34 @@ class WorkspaceRepository(BaseRepository):
     def __init__(self, client: CosmosClient):
         super().__init__(client, config.STATE_STORE_RESOURCES_CONTAINER)
 
-    @property
-    def container(self) -> ContainerProxy:
-        return self._container
+    @staticmethod
+    def _active_workspaces_query():
+        return 'SELECT * FROM c WHERE c.resourceType = "workspace" AND c.isDeleted = false'
+
+    def _get_template_version(self, template_name):
+        wt_repo = WorkspaceTemplateRepository(self._client)
+        template = wt_repo.get_current_workspace_template_by_name(template_name)
+        return template["version"]
 
     def get_all_active_workspaces(self) -> List[Workspace]:
-        query = QueryBuilder().select_active_resources(ResourceType.Workspace).build()
-        workspaces = list(self.container.query_items(query=query, enable_cross_partition_query=True))
-        return workspaces
+        query = self._active_workspaces_query()
+        return self._query(query=query)
 
     def get_workspace_by_workspace_id(self, workspace_id: UUID4) -> Workspace:
-        query = QueryBuilder().select_active_resources(ResourceType.Workspace).with_id(workspace_id).build()
-        workspaces = list(self.container.query_items(query=query, enable_cross_partition_query=True))
-
-        if workspaces:
-            return workspaces[0]
-        else:
+        query = self._active_workspaces_query() + f' AND c.id="{workspace_id}"'
+        workspaces = self._query(query=query)
+        if len(workspaces) != 1:
```

Comment:
In this case I think writing it as 
if not workspaces:
   raise exception
is more elegant to read.